tests/core/data/test_tensordense.py

```diff
@@ -0,0 +1,170 @@
+import numpy as np
+from numpy.testing import assert_almost_equal
+import tensorflow as tf
+import pytest
+
+from qutip.core import data
+from qutip.core.data import dense
+from qutip_tensorflow.core.data import DenseTensor
+
+from . import conftest
+
+
+# Set up some fixtures for automatic parametrisation.
+
+
+@pytest.fixture(
+    params=[
+        pytest.param((1, 5), id="ket"),
+        pytest.param((5, 1), id="bra"),
+        pytest.param((5, 5), id="square"),
+        pytest.param((2, 4), id="wide"),
+        pytest.param((4, 2), id="tall"),
+    ]
+)
+def shape(request):
+    return request.param
+
+
+@pytest.fixture(params=[True, False], ids=["Fortran", "C"])
+def fortran(request):
+    return request.param
```

Comment:
Remove this

Reply:
Actually, do not remove this as `numpy_dense` creates an array that depends on this fixture. For example, it necessary to test that `DenseTensor` is created properly from a Fortran array. Although, I am not sure how important is this.